src/main/src/main/java/org/geoserver/catalog/FeatureTypeInfo.java

```diff
@@ -193,4 +193,13 @@
 	
 	void setCircularArcPresent(boolean arcsPresent);
 
+
+	/**
+	 * Whether querying the full number of features is banned for this feature type.
+	 * 
+	 * @return True if a full feature count should not be attempted, false otherwise
+	 */
+	boolean isTotalCountBanned();
```

Comment:
Primitive... are you sure that old configurations load fine if they don't find this element, and likewise, old REST requests? I guess so, it's probably defaulting to false, but it's important that you double check.